docs/layer2/layer2_support.md

```diff
@@ -0,0 +1,401 @@
+metadata:
+  name: example-packet-machine-template
+spec:
+  template:
+    spec:
+      facility: ny5
+      metro: ny
+      plan: c3.small.x86
+      billingCycle: hourly
+      project: your-packet-project-id
+      sshKeys:
+        - ssh-rsa AAAAB3...your-public-key...
+      operatingSystem: ubuntu_20_04
+      ports:
+       - name: eth1
+          bonded: false
+          layer2: true
+          ip_addresses:
+            - ipAddressReservation: "10.50.10.0/24"
+              vxlan_ids: [1001]
+              gateway: "10.50.10.1"
+
+```
+
+### APIs:
+
+* * * * *
+
+Following are some of the APIs provided by EM, that would be used:
+1. **Convert the port to a layer2 port**:
+
+    a. https://deploy.equinix.com/developers/api/metal/#tag/Ports/operation/convertLayer2
+    b. Endpoint: https://api.equinix.com/metal/v1/ports/{id}/convert/layer-2
+    c. Requied Params : vnid (VLAN ID) 
+
+2. **Assign a port to a virtual network (VLAN)**:
+
+    a. https://deploy.equinix.com/developers/api/metal/#tag/Ports/operation/assignPort
+
+    b. Endpoint: https://api.equinix.com/metal/v1/ports/{id}/assign
+Requied Params : vnid (VLAN ID)
+	c. Type: POST
+    d. Batch Mode
+    ```
+    curl -X POST \
+    -H "Content-Type: application/json" \
+    -H "X-Auth-Token: <API_TOKEN> " \
+    "https://api.equinix.com/metal/v1/ports/{id}/vlan-assignments/batches" \
+    -d '{
+        "vlan_assignments": [
+            {
+                "vlan": "string",
+                "state": "assigned"
+            },
+            {
+                "vlan": "string",
+                "state": "assigned"
+            },
+        ]
+    }'
+    ```
+
+3. **Device Events API**:
+ 	a. Endpoint:  `https://api.equinix.com/metal/v1/devices/<id>/events`
+
+4. **Remove port from the bond**
+    a. Endpoint: 
+    ```
+    curl -X POST \
+        -H "Content-Type: application/json" \
+        -H "X-Auth-Token: <API_TOKEN>" \
+        "https://api.equinix.com/metal/v1/ports/{id}/disbond" \
+        -d '{
+            "bulk_disable": false
+        }'
+    ```
+
+
+### User-Data Script for Network Configuration
+To configure the operating system (OS), create new sub-interfaces for handling VLAN-tagged traffic, and assign IP addresses to those sub-interfaces, a user-data script is required to run at the time of OS boot.
+Below is the user-data script that would be used.
+
+```sh
+#cloud-config
+package_update: true
+package_upgrade: true
+packages:
+  - jq
+  - vlan
+
+write_files:
+  - path: /var/lib/capi_network_settings/final_configuration.sh
+    permissions: '0755'
+    content: |
+      #!/bin/bash
+      set -euo pipefail
+
+      echo "Running final configuration commands"
+      apt-get update -qq
+      apt-get install -y -qq jq vlan
+
+      modprobe 8021q
+      echo "8021q" >> /etc/modules
+
+      # Generate the network configuration and append it to /etc/network/interfaces for each VLAN-tagged sub-interface.
+      cat <<EOL >> /etc/network/interfaces
```

Comment:
We can write out directly to interfaces.d/something.  An example of that here:

https://github.com/equinix-labs/terraform-equinix-metal-hybrid-gateway/blob/main/modules/backend/cloud-config.cfg#L7

Reply:
Err .. nevermind that example.. It was more involved and manipulated cloud-config.d rather than interfaces.d, it also depended on changing the OS to use netplan:

https://github.com/equinix-labs/terraform-equinix-metal-hybrid-gateway/blob/main/modules/backend/pre-cloud-config.sh

That said, we could write out to an interfaces.d/ file directly. This is just a nitpic and there may be reasons why it is better to take the inline approach you offered, to ensure that the modification is made at the right time relative to other operations.

Reply:
I really don't have any opinions on `/etc/network/interfaces` vs `/interfaces.d` directory  😅
If it seems reasonable to write into  `/interfaces.d` , I will try that out once.